### ISSUE TITLE
Can't enter any ESP32 AT-commands on Windows using any terminal (putty, tera term ....)

### ISSUE THREAD

**Issue by tchouzin** [espressif/esp32-at#117]
Hello,
I hope can get your help and I appreciate you in advance.

Can anybody tell me an simple procedure to setup and build AT-command on Windows from scratch? 

I followed the procedure described in the following link without success: https://github.com/espressif/esp32-at 

I also looked at the "issues" section, but could not solve the problem.

Any response would be very helpful.
Thanks

Achille T.

![capture](https://user-images.githubusercontent.com/42009077/43543100-e948c58c-959c-11e8-8d2b-7aca905de350.JPG)
![capture](https://user-images.githubusercontent.com/42009077/43543166-2578f93c-959d-11e8-99f9-1573c0cb9bbe.JPG)

**Comment by JuPrgn** [comment]
Hi you have to configure the UART your are using to send at commands using menuconfig depending on your hardware you will chose UARTx and you will configure the gpio pins. You can also configure \r or \n or both as end of command I had some issue using Putty an I used pqcom on Windows with that terminal you can select which ending you want but don't forget to check the checkbox on this menu that doesn't work if this is not checked.

**Comment by tchouzin** [comment]
Thanks for your answer #JuPrgn. 
I configured the gpio pins and UART and checked that everything was correct. But I still have no expected results.
I have two modules esp32 ( https://learn.adafruit.com/adafruit-huzzah32-esp32-feather/overview and https://www.espressif.com/en/products/hardware/esp32-devkitc/overview ). But I have the same results.

The modules restart in infinite loop when I type the command "make monitor", or when i open the port in another terminal. 

I really need help.

Thank you in advance for giving me some of your time to help me make it happen, because I need it for a school project.

Here are some screenshots of configurations
![capture10](https://user-images.githubusercontent.com/42009077/43687495-049bbccc-98a4-11e8-87da-06e860f46d00.PNG)
![capture11](https://user-images.githubusercontent.com/42009077/43687497-08b934ec-98a4-11e8-99de-6ffd3cb0dd37.PNG)

**Comment by tchouzin** [comment]
Thanks for your answer #JuPrgn.
I configured the gpio pins and UART and checked that everything was correct. But I still have no expected results.
I have two modules esp32 ( https://learn.adafruit.com/adafruit-huzzah32-esp32-feather/overview and https://www.espressif.com/en/products/hardware/esp32-devkitc/overview ). But I have the same results.

The modules restart in infinite loop when I type the command "make monitor", or when i open the port in another terminal.

I really need help.

Thank you in advance for giving me some of your time to help me make it happen, because I need it for a school project.

Here are some screenshots of configurations
![capture10](https://user-images.githubusercontent.com/42009077/43687579-34627c7e-98a5-11e8-944a-6807c2c06cc5.PNG)
![capture11](https://user-images.githubusercontent.com/42009077/43687581-37c568a4-98a5-11e8-93fb-3d9e9fbb2948.PNG)

**Comment by tchouzin** [comment]
Hello everyone,
I keep looking for a solution to my problem specified above. Despite the configurations that are made, I still do not receive anything. I wonder what I missed in this step that seems clear.
I also used the pqcom terminal as suggested by #JuPrgn, but that did not solve the problem.

I ask once again for your help.

Thank you in advance.

Below, some screenshots.
![capture12](https://user-images.githubusercontent.com/42009077/43734671-748b2224-9986-11e8-8fbc-caac3b4a6d23.JPG)

![capture13](https://user-images.githubusercontent.com/42009077/43734693-8c1e47fe-9986-11e8-984a-ca437556d384.JPG)

**Comment by tchouzin** [comment]
Hello @xcguang 
Any help on this?

Thanks

**Comment by xcguang** [comment]
Hi @tchouzin ，
I guess the serial tools do not send “\r\n", please try this tool. 

![image](https://user-images.githubusercontent.com/25979965/43777780-678428d6-9a86-11e8-9514-b77eff6ba6e8.png)

[QCOM_V1.1.zip](https://github.com/espressif/esp32-at/files/2266459/QCOM_V1.1.zip)

**Comment by tchouzin** [comment]
Hello @xcguang ,
Thank you for your help. It works now with the QCOM terminal.

Thanks again.

**Comment by rkashish75** [comment]
Can anyone help with AT commands through Arduino IDE for ESP32?
https://github.com/xcguang can you help?

**Comment by xcguang** [comment]
Hi @rkashish75 What we need to do? We can evaluate it。

**Comment by rkashish75** [comment]
I need to scan continuously for BT devices in vicinity through enquiry scan
and send the MAC address, of the detected devices to a web server
periodically.
We did this using Arduino Mega, ESP01 Wi-Fi and BT HC-05 in master mode.
BT is used for continuous enquiry scan.

**Comment by xcguang** [comment]
Hi, 
Now do you want to replace Arduino Mega, ESP01 Wi-Fi and BT HC-05 with ESP32?
We have not used Arduino, can you provide more detail about the issue?

**Comment by rkashish75** [comment]
Yes. I want to replace the combined hardware with ESP-32.
My problem is that AT commands do not work as smooth and simple as in case
of Arduino Uno/Mega and BT HC-05 combination.
I have installed the ESP_AT Library by Khoi Hoang.
Also did the complete toolchain and tried using the mingw33.exe.
Didn't get the expected response to AT commands. Whereas I tried QCOM V1.1
as suggested by you. Port opens and I get the initial booting information
and few devices are discovered by default.

I need to do continous scanning for nearby BT devices which are
discoverable. I don't need any pairing or handshake. Only enquiry.
Send the collected Mac I'ds periodically to a web server.
